File: src/co/edu/uptc/aj/os/gui/TablePanel.java
package co.edu.uptc.aj.os.gui;

import java.awt.BorderLayout;
import java.util.ArrayList;

import javax.swing.JPanel;
import javax.swing.JScrollPane;
import javax.swing.JTable;
import javax.swing.table.DefaultTableModel;

import co.edu.uptc.aj.os.logic.Process;

public class TablePanel extends JPanel {
    private JTable table;
    private DefaultTableModel dtm;

    public TablePanel() {
        begin();
        setLayout(new BorderLayout());
        addComponents();
    }

    private void addComponents() {
        add(new JScrollPane(table), BorderLayout.CENTER);
    }

    private void begin() {
        String[] titles =
            {"Proceso", "Tamaño", "Prioridad", "T LLegada", "T Espera", "T Inicio", "T Retorno"};
        dtm = new DefaultTableModel(titles, 0);
        table = new JTable(dtm);
        table.setFillsViewportHeight(true);
        table.setEnabled(false);
    }

    public void addRow(Process pro) {
        Object[] row = {pro.getName(), pro.getLength(), pro.getPriority(), pro.getArrival(),
                        pro.getWait(), pro.getStart(), pro.getEnd()};
        dtm.addRow(row);
    }

    public void cleanTable() {
        for (int i = 0; i < table.getRowCount(); i++) {
            dtm.removeRow(i);
            i -= 1;
        }
    }

    public void repaintTable(ArrayList<Process> processes) {
        cleanTable();
        for (Process pro : processes) {
            addRow(pro);
        }
    }
}

File: src/co/edu/uptc/aj/os/gui/TablePanel.kt
package co.edu.uptc.aj.os.gui

import co.edu.uptc.aj.os.logic.Process
import java.awt.BorderLayout
import java.util.ArrayList
import javax.swing.JPanel
import javax.swing.JScrollPane
import javax.swing.JTable
import javax.swing.table.DefaultTableModel

class TablePanel : JPanel() {
    private var table: JTable? = null
    private var dtm: DefaultTableModel? = null
    
    init {
        begin()
        layout = BorderLayout()
        addComponents()
    }
    
    private fun addComponents() {
        add(JScrollPane(table), BorderLayout.CENTER)
    }
    
    private fun begin() {
        val titles = arrayOf(
            "Proceso", "Tamaño", "Prioridad", "T LLegada", "T Espera", "T Inicio", "T Retorno")
        dtm = DefaultTableModel(titles, 0)
        table = JTable(dtm)
        table?.fillsViewportHeight = true
        table?.isEnabled = false
    }
    
    fun addRow(pro: Process) {
        val row =
            arrayOf(pro.name, pro.length, pro.priority, pro.arrival, pro.wait, pro.start, pro.end)
        dtm?.addRow(row)
    }
    
    fun cleanTable() {
        var i = 0
        while (i < (table?.rowCount ?: 0)) {
            dtm?.removeRow(i)
            i -= 1
            i++
        }
    }
    
    fun repaintTable(processes: ArrayList<Process>) {
        cleanTable()
        for (pro in processes) {
            addRow(pro)
        }
    }
}